File: golang/506.relative-ranks.go
func findRelativeRanks(score []int) []string {
	n := len(score)
	places := make([]int, n)
	for i := 0; i < n; i++ {
		places[i] = i
	}

	sort.Slice(places, func(i, j int) bool {
		return score[places[i]] > score[places[j]]
	})

	res := make([]string, n)
	for i := 0; i < n; i++ {
		res[places[i]] = strconv.Itoa(i + 1)
	}

	prize := map[int]string{0: "Gold Medal", 1: "Silver Medal", 2: "Bronze Medal"}
	for i := 0; i < min(3, n); i++ {
		res[places[i]] = prize[i]
	}

	return res
}

File: cpp/506.relative-ranks.cpp
class Solution {
public:
  vector<string> findRelativeRanks(vector<int> &score) {
    int n = score.size();
    vector<string> res(n, "");
    vector<int> places(n);
    iota(places.begin(), places.end(), 0);
    sort(places.begin(), places.end(),
         [&score](int a, int b) -> bool { return score[a] > score[b]; });

    for (int i = 0; i < n; i++) {
      res[places[i]] = to_string(i + 1);
    }

    map<int, string> ranks = {
        {0, "Gold Medal"}, {1, "Silver Medal"}, {2, "Bronze Medal"}};

    for (int i = 0; i < min(3, n); i++) {
      res[places[i]] = ranks[i];
    }

    return res;
  }
};